monkey/infection_monkey/monkey.py

```diff
@@ -155,11 +155,17 @@ def __init__(self, args, ipc_logger_queue: multiprocessing.Queue, log_path: Path
             Path(gettempdir())
             / f"infection_monkey_plugins_{self._agent_id}_{secure_generate_random_string(n=20)}"
         )
-        self._island_address, self._island_api_client = self._connect_to_island_api()
-        self._local_network_interfaces = get_network_interfaces()
-        self._register_agent()
 
         self._operating_system = get_os()
+        self._local_network_interfaces = get_network_interfaces()
```

Comment:
I think it would be better to refactor everything using `self._local_network_interfaces` to use `self._local_machine_info.network_interfaces`

Reply:
Good call. I did the same for `self._operating_system`